### ISSUE TITLE
Any Updates?

### ISSUE THREAD

**Issue by GageBachik** [Hotrian/OpenVRDesktopDisplayPortal#60]
Been awhile since the last release and a lot of updates to steamVR, graphics drivers, and games since then. Wondering if you're still working on this as some games like: the Brookhaven experiment cause memory leaks when using this app.

**Comment by Hotrian** [comment]
Yes I am still working on this. I have taken the source closed source for the time being, I went over my reasoning [in this post on Reddit](https://www.reddit.com/r/Vive/comments/5d4576/so_i_rigged_up_a_solution_for_keeping_an_eye_on/da2au3v/?context=10000). I plan to leave the current source available, but I'm planning on posting the new version to Steam soon. I've been in talks with Valve so whenever they get moving forward on that end it should be available on Steam. If they don't let me post it on Steam for whatever reason, I'll be releasing my local source as builds here on GitHub, but I won't be updating the public source code for the foreseeable future. 

I have not posted any new builds recently because this is made in Unity and written on C#. C# is a managed language, so it's pretty easy to reverse engineer, and I don't want to give my competition any legs up while my Steam application is pending.

Can you be more specific about the Brookhaven Experiment having memory leaks? I am not doing anything funky here - I am using the OpenVR API to draw the Overlay, so if there are memory leaks with BE being caused this is an issue with BE or SteamVR, and not an issue with OVRDDP itself - if however OVRDDP is the one having memory leaks, then that is certainly something I can look into.

Edit: To give you an idea of the improvement that's been made, the current public release can only capture about a 360p to a 480p area, depending on hardware, at 24FPS or greater which is required for video capture and playback to appear smooth. My local source can easily capture my full monitor (2560x1440) at 150fps up to around 500fps if things are changing fast enough - the API I'm working on only updates when something on the screen changes so it uses a lot less CPU and can still capture insanely faster because it is GPU accelerated - though it requires Windows 8 or above to function. I've been working on a bunch of other new things as well, but having HD capture is the most important change IMO between the public release and my local source.

Also, sorry for the late reply. For some reason GitHub did not notify me of this issue being posted until now.

**Comment by GageBachik** [comment]
Hey! First off thanks so much for taking the time to write this response!

I took the time to read the full reddit post and edits. I completely respect the decision to take this close source and put it up on steam. Honestly my stream depends on this software so I don't mind paying for it if it does the job correctly.

I'll look more into exactly where the leaks are coming from and submit a bug request to BE based on the results.

Anyways the improvements obviously huge for me as I actually Stream VR and this app is essential for chat and what not. Please hit up this issue with the steam link when its available (or product page depending on steam - I know how they can be -) and I'll be sure to be one of your first customers / drop a review.

-Thanks again!

**Comment by Hotrian** [comment]
Just an update about the Steam version:

Still waiting for the Steam onboarding process. I was first in contact with Valve back in November when they told me I would be getting on Steam. I've been told multiple times I should be getting on Steam, but then I hear nothing back from them :/. The most recent time was just a few days ago, so hopefully OVRDDP will be added to Steam shortly..

**Comment by hammerschaedel** [comment]
hmm, shouldn´t it be easier to get on steam with a VR Program? maybe because of the Holidays?

**Comment by Hotrian** [comment]
It certainly should be! I keep talking to this one guy at Valve, and he keeps CCing this other guy whose supposedly in charge of the Steam onboarding process, telling him to get me on Steam, but that other guy never writes me back!

I've been emailing them since November :(. Once they add me on Steam I just need to work out the Steamworks API, integrate it, and finish fixing up the UI, and then OVRDDP will be ready for launch! It's very stable so far in my testing, insanely faster and less resource intensive, and ready for a public launch as far as I'm concerned - though it does need a prettier UI setup :). 

I definitely have plans to add more features, but the first version is ready to go, aside from the aforementioned changes. 

The only major downside now is that it's only compatible with Windows 8 and up due to the new API not being available in Windows 7 and below.

**Comment by hammerschaedel** [comment]
hmm...maybe you should go then the way over Greenlight... you have a must-have Tool for the Vive und with the Help of the Reddit Community this should not be a Problem. i´m also already promoting this at the german vrforum ;)

**Comment by Hotrian** [comment]
I'm thinking that may be the case. Still waiting on Valve to add me. I just sent them another email, but if they don't add me soon I'll look into trying to Greenlight it. I expect that would take even longer since they still have to add me (Greenlight doesn't happen automatically), but maybe with enough support I will get added faster :/.

**Comment by Hotrian** [comment]
Update! Valve is finally adding me :). It will still be some time before I am ready for launch (have to implement Steamworks, bug test, etc) but it is definitely coming now.

**Comment by this-is-neat** [comment]
Made an account just to say how cool this is! Can't wait to see it on steam. Keep us updated on how it goes. Thanks for the awesome work.

**Comment by Hotrian** [comment]
It's official! I was accepted and am now a Streamworks Developer :).

**Comment by zombicadavre** [comment]
Can't wait to see results :) Well done!

**Comment by hammerschaedel** [comment]
yay...very cool..gratz

**Comment by zombicadavre** [comment]
Hotrian, do you have any news? :)

**Comment by Hotrian** [comment]
I'm still actively working to prepare for launch :).

Unfortunately I'm not sure what I can share right now. I'm under Steam's NDA so I don't want to get in trouble disclosing any part of the process. What I _can_ say is that things are moving forward and I'm hoping to have OVRDDP for sale within the next month or so :).

**Comment by hammerschaedel** [comment]
excellent... i switched in the meantime to the Win10 Fastring to try the new Gamemode with ed.....which makes the Actual Version of ODDP very laggy and nearly unusable. I hope that the new faster version runs better and counting the days to release^^

**Comment by DemmyDemon** [comment]
I keep throwing money at my HMD, but nothing is happening.

**Comment by ragesaq** [comment]
I keep refreshing the source page hoping to see the announcement! :)

**Comment by Hotrian** [comment]
Update! 

Still working towards release. Waiting for Valve to do some stuff now, and it should (hopefully) be released by Friday!

I got all the things I needed finished and I'm just waiting for final approval now before it goes live :).

**I have renamed OpenVRDesktopDisplayPortal to _OVRdrop_**

The store page should be [here](http://store.steampowered.com/app/586210) once it goes live.

I have a new Issue/Feature Tracker only repo setup [here](https://github.com/Hotrian/OVRdrop-Public/) that I'll mark as public once OVRdrop is released.

Thanks for the support guys! I couldn't have come this far without the community requesting features and reporting bugs.

**Comment by folkdeezy** [comment]
this is gonna be sick!

**Comment by Hotrian** [comment]
OVRdrop's store page [is now live](http://store.steampowered.com/app/586210)!

Still planning for a Friday release barring unforeseen circumstances.

**Comment by zombicadavre** [comment]
Well done!

**Comment by Hotrian** [comment]
OVRdrop is now on sale! $12.74 until April 7th :).

**Comment by zombicadavre** [comment]
I tried OVRdrop. I already have a problem, I'll make a post on the tracker

**Comment by Hotrian** [comment]
Fuuuu!

Edit: Any update on this @zombicadavre ? I don't see anything on the issue tracker. I'm thinking maybe you just had an issue getting setup and it wasn't actually a bug? OVRdrop isn't entirely streamlined yet unfortunately, and can be a bit confusing at first.

**Comment by zombicadavre** [comment]
I gave up because of  late night :) had to get some sleep 

I did a little research on what happened, and it was capture problem from monitor doesn't connected to same gpu as hmd, So you're right it is setup problem or feature but not a bug. I used recently UVRO with same config and didn't expect such behavior from OVRdrop, it's my fault.

BTW. There were several stops of working with "Not responding message".

**Comment by Hotrian** [comment]
Weird, I've never had any "Not responding" issues. Can you send me your `output_log.txt` from the Data folder and also your SteamVR System Report?

Unfortunately I only have a single GTX 1080 and a single R7265 in a different PC, so I can't test for multi-gpu issues and I'm not sure any of my beta testers tested against that either.

**Comment by zombicadavre** [comment]
Here it is

[output_log.txt](https://github.com/Hotrian/OpenVRDesktopDisplayPortal/files/887287/output_log.txt)
[steamvr report.txt](https://github.com/Hotrian/OpenVRDesktopDisplayPortal/files/887288/steamvr.report.txt)
